unison-cli/src/Unison/CommandLine/OutputMessages.hs

```diff
@@ -346,6 +349,93 @@ notifyNumbered o = case o of
           <> IP.makeExample' IP.viewReflog
           <> "to undo this change."
 
+showListEdits :: Patch -> PPE.PrettyPrintEnv -> (P.Pretty P.ColorText, NumberedArgs)
+showListEdits patch ppe =
+  ( P.sepNonEmpty
+      "\n\n"
+      [ if null types
+          then mempty
+          else
+            "Edited Types:"
+              `P.hang` P.column2 typeOutputs,
+        if null terms
+          then mempty
+          else
+            "Edited Terms:"
+              `P.hang` P.column2 termOutputs,
+        if null types && null terms
+          then "This patch is empty."
+          else
+            tip . P.string $
+              "To remove entries from a patch, use "
+                <> IP.deleteTermReplacementCommand
+                <> " or "
+                <> IP.deleteTypeReplacementCommand
+                <> ", as appropriate."
+      ],
+    numberedArgsCol1 <> numberedArgsCol2
+  )
+  where
+    typeOutputs, termOutputs :: [(P.Pretty CT.ColorText, P.Pretty CT.ColorText)]
+    numberedArgsCol1, numberedArgsCol2 :: NumberedArgs
+    -- We use the output of the first column's count as the first number in the second
+    -- column's count. Laziness allows this since they're used independently of one another.
+    (((typeOutputs, termOutputs), (lastNumberInFirstColumn, _)), (numberedArgsCol1, numberedArgsCol2)) =
```

Comment:
Apologies for using the fixed-point of a State monad here, I don't love it, but I tried doing the equivalent version by unzipping, collecting, counting, then re-zipping and it ended up much more complicated.

We could reduce complexity slightly (i.e. avoid the fixed point) by switching the numbering scheme to this:

```
.thing> view.patch

  Edited Types:
    1. Unit         -> 2. X
    3. Z#d97e0jhkmd -> 4. Z

  Edited Terms:
    5. #jk19sm5bf8 -> 6. x
    7. #tufqout1uq -> 8. z
```

But I think that output is slightly less useful for ranges.

IMO a comment noting the fixed point is sufficient, but if folks are opposed and are okay with the other numbering format I wouldn't be opposed to switching 👍🏼

Reply:
If we decide that ranges aren't useful here anyway, then go ahead with whatever makes the code simpler, but at the moment I truly don't have any strong opinion.